addons/purchase/purchase.py

```diff
@@ -736,19 +735,19 @@ def propagate_cancels(self):
 
         return result
 
-    @api.model
-    def _run(self, procurement):
-        if procurement.rule_id and procurement.rule_id.action == 'buy':
-            return procurement.make_po()
-        return super(ProcurementOrder, self)._run(procurement)
+    @api.multi
+    def _run(self):
+        if self.rule_id and self.rule_id.action == 'buy':
```

Comment:
It's more explicit but doesn't actually matter, attribute access (`.rule_id`) will implicitly `ensure_one`.

However the pre-check on `self.rule_id` is unnecessary: if the m2o is not set it'll just return an empty recordset, and `.action` will return `False.